Framework/Kernel/inc/MantidKernel/VMD.h

```diff
@@ -234,12 +234,12 @@ template <typename TYPE = double> class DLLExport VMDBase {
     boost::split(strs, str, boost::is_any_of(", "));
 
     std::vector<TYPE> vals;
-    for (size_t d = 0; d < strs.size(); d++) {
-      if (!strs[d].empty()) {
+    for (const auto &str : strs) {
```

Comment:
I just realized that the argument of the function is also called `str`. If this stays as a for-loop I'd suggest renaming one of the two to avoid confusion.